src/fsharp/lex.fsl

```diff
@@ -703,11 +729,37 @@ rule token args skip = parse
 
  | ">]" { GREATER_RBRACK }
 
- | "{" { LBRACE }
+ | "{"
+     {
+       if args.interpolatedStringNesting > 0 then
+           args.interpolatedStringNesting <- args.interpolatedStringNesting + 1
+       LBRACE
+     }
 
  | "|" { BAR }
 
- | "}" { RBRACE }
+ | "}" 
+     { 
+       // We encounter a '}' in the expression token stream.  First check if we're in an interpolated string expression
+       // and continue the string if necessary
+       if args.interpolatedStringNesting = 1 && args.interpolatedStringStyle.IsSome then
```

Comment:
Should check for deeper nested conditions.
This should be why `$"{$"{1}"}"` fails.

Reply:
last year I was stuck exactly here :p

Reply:
instead of a single mutable counter, we could have a stack. each nested interp string requires a new context.
the outer counter is still needed, because we will have braces in interpolated code for records etc.

Reply:
see https://github.com/dotnet/fsharp/pull/8997

Reply:
Thanks, good work!